cli/src/main/java/org/jboss/as/cli/accesscontrol/BaseAccessRequirement.java

```diff
@@ -34,17 +34,16 @@
  */
 public abstract class BaseAccessRequirement implements AccessRequirement, CliEventListener {
 
-    protected Logger log = Logger.getLogger(getClass());
-    protected final boolean traceEnabled = log.isTraceEnabled();
-
     private Boolean satisfied;
 
     @Override
     public boolean isSatisfied(CommandContext ctx) {
         if(satisfied == null) {
             satisfied = checkAccess(ctx);
-            if(traceEnabled) {
-                log.trace(toString() + " " + satisfied);
+            if(satisfied) {
+                CliLogger.ROOT_LOGGER.accessRequirementSatisfied(toString());
+            } else {
+                CliLogger.ROOT_LOGGER.accessRequirementNotSatisfied(toString());
             }
```

Comment:
This should probably be
```java
if (CliLogger.ROOT_LOGGER.isTraceEnabled()) {
    if(satisfied) {
        CliLogger.ROOT_LOGGER.accessRequirementSatisfied(toString());
    } else {
        CliLogger.ROOT_LOGGER.accessRequirementNotSatisfied(toString());
    }
}
```

The `toString()` may be expensive.

Reply:
I guess the other option would be to use.
```java
if(satisfied) {
    CliLogger.ROOT_LOGGER.accessRequirementSatisfied(this);
} else {
    CliLogger.ROOT_LOGGER.accessRequirementNotSatisfied(this);
}
```
You'd just need to change the `accessRequirementNotSatisfied()` method to accept a `BaseAccessRequirement` or just `Object`. Either should be fine though.